virtualizarr/readers/tiff.py

```diff
@@ -1,79 +1,179 @@
-import warnings
-from pathlib import Path
-from typing import Hashable, Iterable, Mapping, Optional
+from __future__ import annotations
 
-from xarray import Dataset, Index
+import dataclasses
```

Comment:
This doesn't seem to be used in this file? (and the linter should have detected that...)